packages/errors/src/index.ts

```diff
@@ -55,8 +55,8 @@ export const isWorkflowError = (error: Error) => error.name.includes('WorkflowEr
 /**
  * Returns true if the error is a DynamoDB conditional check exception.
  */
-export const isConditionalCheckException = (error: ErrorWithOptionalCode) =>
-  error.name === 'ConditionalCheckFailedException' || error.code === 'ConditionalCheckFailedException';
+export const isConditionalCheckException = (error: any) =>
+  error.name === 'ConditionalCheckFailedException';
```

Comment:
This is a tough one. I'm fairly certain the issue here is that the library throwing this exception is `@aws-sdk/lib-dynamodb`, not one of the `client-x` packages. The clients appear to support `instanceof` via the expected chain of inheritance, the lib export... may not.

Switching this over to `instanceof` like the others does not work. The error LOOKS like it's a `ConditionalCheckFailedException` but the properties look different than other errors and, again, `instanceof` does not work despite the class presenting as we'd expect.

I've chosen to fall-back to the supported `name` property in this case. According to their docs:

"This change is backward compatible—all the existing error handling code will still work. You can still assert the error by the name value."

Reply:
Also I _think_ we want the error type here to be `any` because of the above. We want to be able to inspect an object that looks like a `ConditionalCheckFailedException` class instance but doesn't respond to `instanceof`.

Reply:
Yeah, so the worst case result here is still supported. It's not the most optimal/consistent but it won't cause problems... allegedly.